launcher/src/main/java/org/apache/spark/launcher/OutputRedirector.java

```diff
@@ -61,6 +62,10 @@ private void redirect() {
       while ((line = reader.readLine()) != null) {
         if (active) {
           sink.info(line.replaceFirst("\\s*$", ""));
+          if (error == null && containsIgnoreCase(line, "Error") || containsIgnoreCase(line,
```

Comment:
nit: group the conditions so people don't have to remember how precedence work. Also it looks better if you keep the whole call to `containsIgnoreCase` in the same line.

Wouldn't you want to keep the last error, though, instead of the first one?